sound/soc/sof/sof-client-probes.c

```diff
@@ -301,7 +306,11 @@ static int sof_probes_compr_shutdown(struct snd_compr_stream *cstream,
 	priv->extractor_stream_tag = SOF_PROBES_INVALID_NODE_ID;
 	snd_compr_free_pages(cstream);
 
-	return ops->free(cdev, cstream, dai);
+	ret = ops->free(cdev, cstream, dai);
+
+	sof_client_core_module_put(cdev);
```

Comment:
same I don't get this at all.

Reply:
We are actively using the core during the time when the stream is running and we need our parent to be around for the duration. Outside of that time the parent can be removed, which will destroy the device for the probes and other auxiliary clients.